core/src/main/java/org/jdbi/v3/core/internal/OnDemandHandleSupplier.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.internal;
+
+public interface OnDemandHandleSupplier {}
```

Comment:
Oh god, does this only exist because LazyHandleSupplier is package-private and not visible to CreateSqlObjectHandler, and can't be made public because it can't move to `internal` because it needs to call package-private methods on `Handle` that can't be made public?